### DISCUSSION TITLE
customise AdGuardHome.yaml?

### DISCUSSION THREAD

**Discussion by Rusketh** [hassio-addons/addon-adguard-home#365] [Q&A] [answered]
I find myself in need of access to settings that are not available from the UI.
Is there anyway of adding a node to the Addons config file that then injects into the AdGuardHome.yaml
like

```
!options.yml
```

Not sure that is how you would go about it, might need some form of python or bash script to do some trickery on boot & create a new config file on the fly.

**Comment by frenck** [comment, marked as answer]
The add-on doesn't support that. If you are in need of such a level of control/customization, this add-on isn't for you.

**Comment by Rusketh** [comment]
Would you be open to the option of this if I where to find a way to do this?

**Reply by frenck** [reply]
I don't think so, mainly because there is no clean way to do it.

Additionally, I don't think we should have to support every use case, mostly the common ones are importantly. With that in mind, if you need to do this, it ain't a common use case.

**Reply by Rusketh** [reply]
Its more so that I want to be able to redirect blocked websites to a custom denied page on the network.
Also allowing my reverse proxy server to work would be nice.
I guess I shall however look into an alternative set up as suggested.

**Reply by LordNex** [reply]
That's something better left to an appliance like a Firewalla Gold can do. Additionally, depending on wifi access points, you can do some IDS on there as well. I'm running Aruba IAPs in a cluster and all my guest traffic that is denied is blocked at the access point so I don't have all that traffic traversing the network. 

AdGuard isn't the same as something like Firewalla or Websense where you can define categories of specific sites like porn and block it as a whole

**Comment by diamant-x** [comment]
I arrived to this request in need of something similar. https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file

In my case the scenario is a bit simpler. I use chrony (hassio-addon) for hosting NTP server from HA. However, i'm not sure when Adguard's DHCP server broadcast its settings whether it is setting such as the NTP server that clients should configure for. So I see that although DHCP should be able to tell all clients what NTP Server they should use, they don't because such field is to be configured in the yaml file as per https://github.com/AdguardTeam/AdGuardHome/wiki/DHCP#config-4 and https://github.com/AdguardTeam/AdGuardHome/issues/1585#issuecomment-619042592 because both Chrony and Adguard Home are standard hassio-addons it would be nice to be able to add it in the configuration.

**Comment by frenck** [comment]
Sorry, if you are in need for such things, this add-on is not for you.
